### PR TITLE
Added OLA management

### PR DESCRIPTION
**Pull request by Fry-4TF1V**
Added 4 modules to manage OVHcloud OLA for Baremetal :
* dedicated_server_ola_configure.py : configure OLA for all network interfaces of a Baremetal
* dedicated_server_ola_configure_wait.py : wait for the completion of the return task from dedicated_server_ola_configure.py
* dedicated_server_ola_unconfigure.py : unconfigure OLA for all network interfaces of a Baremetal
* dedicated_server_ola_unconfigure_wait.py : wait for the completion of the return task from dedicated_server_ola_unconfigure.py

### CODE REVIEW

**Comment by Skunnyk** [comment]
Hi, thank you for your MR :) Can you please fix the linter issues ?

Side question: how to you manage the network configuration on OS side to match link aggregation ? The chicken egg problem: you need to configure bonding on the OS to work with link aggregation, but once you enable link aggregation, you can't connect anymore to the OS :)
You also need to disable OLA to enter in rescue mode or do a reinstall, right ?

**Comment by Fry-4TF1V** [comment]
Hi,

I reviewed and fix the linter issues. Can you try the merge again ?

For the chicken egg problem, I had in mind to setup the private bonding on OS side through the initial public access but not apply it directly. Instead I would force a reboot (or something similar). Then I would go to the OLA config.
For the rescue mode and reinstall, yes it is still required to unconfigure OLA for those operations.

**Review by Skunnyk — APPROVED** [review]

**Review by carrouet — APPROVED** [review]

File: plugins/modules/dedicated_server_ola_configure.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+from ansible.module_utils.basic import AnsibleModule
+
+DOCUMENTATION = '''
+---
+module: dedicated_server_ola_configure
+short_description: Configure all network interfaces in OLA mode
+description:
+    - Configure all network interfaces in an OVHcloud Link Aggregation mode to switch to full network private mode (vrack)
+author: OVHcloud Professional Services
```
Comment (Skunnyk) [inline plugins/modules/dedicated_server_ola_configure.py:15]: You work at OVHCloud ? ;)

**Review by tortuegenialez — APPROVED** [review]

File: plugins/modules/dedicated_server_ola_configure.py (hunk 1 above)
Comment (Fry-4TF1V) [reply plugins/modules/dedicated_server_ola_configure.py:15]: Yes !

File: plugins/modules/dedicated_server_ola_unconfigure_wait.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+# -*- coding: utf-8 -*-
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.module_utils.basic import AnsibleModule
+
+DOCUMENTATION = '''
+---
+module: dedicated_server_ola_unconfigure_wait
```
Comment (Skunnyk) [inline plugins/modules/dedicated_server_ola_unconfigure_wait.py:11]: isn't this the exact same code as `dedicated_server_ola_configure_wait` ? can''t this be factorised ?

**Pull request closed without merging**